### PR TITLE
mocha: Add constants to Suite and Runner

### PR DESCRIPTION
**Pull request by lukeapage**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mochajs/mocha/blob/master/lib/runner.js#L46 and https://github.com/mochajs/mocha/blob/master/lib/suite.js#L552
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mocha/index.d.ts [hunk 1]
```diff
@@ -1701,6 +1739,8 @@ declare namespace Mocha {
         private _onlyTests;
         private _onlySuites;
 
+        static constants: SuiteConstants;
```
Comment (cspotcode) [inline types/mocha/index.d.ts]: Should this be marked `readonly` or am I being paranoid?  (same for `Runner` constants)

**Review by cspotcode — CHANGES_REQUESTED** [review]
One tiny question for either the author or other reviewers; otherwise looks good to me.

File: types/mocha/index.d.ts (hunk 1 above)
Comment (lukeapage) [reply types/mocha/index.d.ts]: @cspotcode I made them readonly

**Review by cspotcode — APPROVED** [review]
:shipit:

**Pull request merged**